### ISSUE TITLE
Deserialization state with ZonedDateTime property fails with exception

### ISSUE THREAD

**Issue by awesometown** [electronicarts/orbit-dynamodb#7]
If the state object serialized to Dynamo contains a ZonedDateTime property, Jackson will fail to deserialize it back out of DynamoDB with the exception:

```
com.fasterxml.jackson.databind.JsonMappingException: Can not construct instance of java.time.ZonedDateTime: no suitable constructor found, can not deserialize from Object value (missing default constructor or creator, or perhaps need to add/enable type information?)
```
I think the solution is just to register the JavaTimeModule from `jackson-datatype-jsr310` when creating an instance of ObjectMapper, and I'm happy to submit a pull request to do this but wanted to run a couple options by you first:

The easiest solution is just to register the module in `DynamoDBConnection.initializeMapper()` and be done with it. I don't *think* there's a downside here as I don't see a way that people could have already persisted ZonedDateTime objects that this would somehow affect.

Alternatively, the configuration could be modified to allow users to provide a custom ObjectMapper instance, or provide a hook to customize the instance that `DynamoDBConnection` creates given the amount of other config `initializeMapper` does, it seems like a customization hook might be the better route.

**Comment by JoeHegarty** [comment]
I agree that adding a hook to allow an application to customize the the mapper seems like the best solution.

**Comment by awesometown** [comment]
Submitted a pull request: https://github.com/orbit/orbit-dynamodb/pull/8

**Comment by JoeHegarty** [comment]
Your PR is included in 0.9.0 - https://github.com/orbit/orbit-dynamodb/releases/tag/v0.9.0
Will be available in Maven Central shortly.

Thanks for your contribution!
